python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -196,10 +206,97 @@ async def stop_sending_telemetry(self) -> None:
                 msg="Not started yet. Ignoring stop command.",
                 response_code=ResponseCode.NOT_STARTED,
             )
-        await self.disconnect_devices()
+
+        while self._devices:
+            device: BaseDevice = self._devices.pop(-1)
+            self.log.debug(f"Closing {device} device with name {device.name}")
+            await device.close()
+
         self._started = False
 
-    @abstractmethod
-    async def disconnect_devices(self) -> None:
-        """Loop over the configuration and start all devices."""
-        raise NotImplementedError
+    def get_device(
```

Comment:
This creates a new device rather than obtaining a cached device. If that is how you anticipate subclasses working then please consider naming this `make_device`.

I am surprised by this behavior because you have an instance variable `_devices`, but this method seems to not interact with it in any way.